.local/share/man/man1/awww.1
.\" Generated by scdoc 1.11.5
.\" Complete documentation for this program is not available as a GNU info page
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.nh
.ad l
.\" Begin generated content:
.TH "awww" "1" "2026\-08\-21"
.PP
.SH NAME
awww \- An Answer to your Wayland Wallpaper Woes
.PP
.SH SYNOPSIS
\fBawww\fR <COMMAND>
.PP
.SH COMMANDS
.PP
\fBclear\fR
.RS 4
Fills the specified outputs with the given color
.PP
.RE
\fBrestore\fR
.RS 4
Restores the last displayed image on the specified outputs
.PP
.RE
\fBclear\-cache\fR
.RS 4
Deletes the \fBawww\fR cache directory
.PP
.RE
\fBimg\fR
.RS 4
Sends an image (or animated gif) for the daemon to display
.PP
.RE
\fBkill\fR
.RS 4
Kills the daemon
.PP
.RE
\fBquery\fR
.RS 4
Asks the daemon to print output information (names and dimensions)
.PP
.RE
\fBhelp [COMMAND]\fR
.RS 4
Print help or the help of the given command
.PP
.RE
.SH OPTIONS
.PP
\fB\-h\fR, \fB\-\-help\fR
.RS 4
Print help (see a summary with '\&\-h'\&)
.PP
.RE
\fB\-V\fR, \fB\-\-version\fR
.RS 4
Print  version
.PP
.RE
.SH DESCRIPTION
\fBawww\fR is a wallpaper manager that lets you change what your monitors display as
a background by controlling the \fBawww\-daemon\fR at runtime.\&
.PP
It supports animated gifs and putting different stuff in different monitors.\& I
also did my best to make it as resource efficient as possible.\&
.PP
To start, begin by running \fBawww\-daemon\fR.\& That will set up the \fBawww\-daemon\fR.\&
Then, you can send images to be displayed with \fBawww img\fR.\& To kill the daemon,
use \fBawww kill\fR.\&
.PP
\fBNote that awww only works in a compositor that implements the layer\-shell
protocol\fR.\& Typically, \fIwlr\-roots\fR based compositors.\&
.PP
.SH FILES
\fBawww\fR will create the following files in your system:
.RS 4
.PD 0
.IP \(bu 4
A socket in \fI${XDG_RUNTIME_DIR}/${WAYLAND_DISPLAY}\-awww\-daemon.\&<specified
.PD
.RE
namespace>.\&socket\fR,  or
.RS 4
\fI/tmp/awww/${WAYLAND_DISPLAY}\-awww\-daemon.\&<specified namespace>.\&socket\fR, if
$XDG_RUNTIME_DIR is not set.\&
.PD 0
.IP \(bu 4
Cache files in \fI$XDG_CACHE_HOME/awww\fR or \fI$HOME/.\&cache/awww\fR if
$XDG_CACHE_HOME does not exist.\& These are used to set the wallpaper to the
previous image when a monitor is (re)connected or turned on.\&
.PD
.PP
.RE
.SH SEE ALSO
\fBawww\-daemon\fR(1) \fBawww\-clear\fR(1) \fBawww\-img\fR(1) \fBawww\-kill\fR(1) \fBawww\-query\fR(1)


.local/share/bash-completion/completions/awww.bash
_awww() {
    local i cur prev opts cmd
    COMPREPLY=()
    if [[ "${BASH_VERSINFO[0]}" -ge 4 ]]; then
        cur="$2"
    else
        cur="${COMP_WORDS[COMP_CWORD]}"
    fi
    prev="$3"
    cmd=""
    opts=""

    for i in "${COMP_WORDS[@]:0:COMP_CWORD}"
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="awww"
                ;;
            awww,clear)
                cmd="awww__subcmd__clear"
                ;;
            awww,clear-cache)
                cmd="awww__subcmd__clear__subcmd__cache"
                ;;
            awww,help)
                cmd="awww__subcmd__help"
                ;;
            awww,img)
                cmd="awww__subcmd__img"
                ;;
            awww,kill)
                cmd="awww__subcmd__kill"
                ;;
            awww,pause)
                cmd="awww__subcmd__pause"
                ;;
            awww,query)
                cmd="awww__subcmd__query"
                ;;
            awww,restore)
                cmd="awww__subcmd__restore"
                ;;
            awww,toggle)
                cmd="awww__subcmd__toggle"
                ;;
            awww,unpause)
                cmd="awww__subcmd__unpause"
                ;;
            awww__subcmd__help,clear)
                cmd="awww__subcmd__help__subcmd__clear"
                ;;
            awww__subcmd__help,clear-cache)
                cmd="awww__subcmd__help__subcmd__clear__subcmd__cache"
                ;;
            awww__subcmd__help,help)
                cmd="awww__subcmd__help__subcmd__help"
                ;;
            awww__subcmd__help,img)
                cmd="awww__subcmd__help__subcmd__img"
                ;;
            awww__subcmd__help,kill)
                cmd="awww__subcmd__help__subcmd__kill"
                ;;
            awww__subcmd__help,pause)
                cmd="awww__subcmd__help__subcmd__pause"
                ;;
            awww__subcmd__help,query)
                cmd="awww__subcmd__help__subcmd__query"
                ;;
            awww__subcmd__help,restore)
                cmd="awww__subcmd__help__subcmd__restore"
                ;;
            awww__subcmd__help,toggle)
                cmd="awww__subcmd__help__subcmd__toggle"
                ;;
            awww__subcmd__help,unpause)
                cmd="awww__subcmd__help__subcmd__unpause"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        awww)
            opts="-h -V --help --version clear restore clear-cache img toggle pause unpause kill query help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__clear)
            opts="-a -n -o -h --all --namespace --outputs --help [COLOR]"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --namespace)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -n)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --outputs)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -o)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__clear__subcmd__cache)
            opts="-h --help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__help)
            opts="clear restore clear-cache img toggle pause unpause kill query help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__help__subcmd__clear)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__help__subcmd__clear__subcmd__cache)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__help__subcmd__help)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__help__subcmd__img)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__help__subcmd__kill)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__help__subcmd__pause)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__help__subcmd__query)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__help__subcmd__restore)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__help__subcmd__toggle)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__help__subcmd__unpause)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__img)
            opts="-a -o -n -f -t -h --all --outputs --namespace --no-cache --no-resize --resize --crop-gravity --fill-color --filter --transition-type --transition-step --transition-duration --transition-fps --transition-angle --transition-pos --transition-bezier --transition-wave --invert-y --help <IMAGE>"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --outputs)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -o)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --namespace)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -n)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --resize)
                    COMPREPLY=($(compgen -W "no crop fit stretch" -- "${cur}"))
                    return 0
                    ;;
                --crop-gravity)
                    COMPREPLY=($(compgen -W "top-left top top-right left center right bottom-left bottom bottom-right" -- "${cur}"))
                    return 0
                    ;;
                --fill-color)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --filter)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -f)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --transition-type)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -t)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --transition-step)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --transition-duration)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --transition-fps)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --transition-angle)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --transition-pos)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --transition-bezier)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --transition-wave)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__kill)
            opts="-a -n -h --all --namespace --help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --namespace)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -n)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__pause)
            opts="-a -n -h --all --namespace --help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --namespace)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -n)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__query)
            opts="-a -j -n -h --all --json --namespace --help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --namespace)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -n)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__restore)
            opts="-a -n -o -h --all --namespace --outputs --help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --namespace)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -n)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --outputs)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -o)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__toggle)
            opts="-a -n -h --all --namespace --help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --namespace)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -n)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        awww__subcmd__unpause)
            opts="-a -n -h --all --namespace --help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --namespace)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -n)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

if [[ "${BASH_VERSINFO[0]}" -eq 4 && "${BASH_VERSINFO[1]}" -ge 4 || "${BASH_VERSINFO[0]}" -gt 4 ]]; then
    complete -F _awww -o nosort -o bashdefault -o default awww
else
    complete -F _awww -o bashdefault -o default awww
fi
